core/object.h

```diff
@@ -531,6 +531,12 @@ class Object {
 	void add_change_receptor(Object *p_receptor);
 	void remove_change_receptor(Object *p_receptor);
 
+// TODO: ensure 'this' is never NULL since it's UB, but by now, avoid warning flood
+#ifdef __clang__
```

Comment:
Clang complains that 'this' can never be null in well-defined C++. And you get a flood of messages about that during compilation. I thought that instead of disabling the warning globally, it'd be better to do it just here and add a note about why and a 'to-do'. All as a transient solution until someone works out whether checking 'this' for null is actually needed or reliable, or audits the whole codebase to ensure that can never happen. The thing is that if it's UB, the compiler is allowed to remove it altogether.